sys/include/net/ng_rpl/structs.h

```diff
@@ -0,0 +1,234 @@
+    uint8_t reserved;   /**< reserved */
+} ng_rpl_dis_t;
+
+/**
+ * @brief Destination Advertisement Object
+ * @see <a href="https://tools.ietf.org/html/rfc6550#section-6.4">
+ *          Destination Advertisement Object
+ *      </a>
+ */
+typedef struct __attribute__((packed)) {
+    uint8_t instance_id;        /**< id of the instance */
+    uint8_t k_d_flags;          /**< K and D flags */
+    uint8_t reserved;           /**< reserved */
+    uint8_t dao_sequence;       /**< sequence of the DAO, needs to be used for DAO-ACK */
+    ng_ipv6_addr_t dodag_id;    /**< id of the DODAG */
+} ng_rpl_dao_t;
+
+/**
+ * @brief Destination Advertisement Object Acknowledgement
+ * @see <a href="https://tools.ietf.org/html/rfc6550#section-6.5">
+ *          Destination Advertisement Object Acknowledgement
+ *      </a>
+ */
+typedef struct __attribute__((packed)) {
+    uint8_t instance_id;        /**< id of the instance */
+    uint8_t d_reserved;         /**< if set, indicates that the DODAG id should be included */
+    uint8_t dao_sequence;       /**< sequence must be equal to the sequence from the DAO object */
+    uint8_t status;             /**< indicates completion */
+    ng_ipv6_addr_t dodag_id;    /**< id of the DODAG */
+} ng_rpl_dao_ack_t;
+
+/**
+ * @brief DODAG Information Solicitation
+ * @see <a href="https://tools.ietf.org/html/rfc6550#section-6.7.7">
+ *          RPL Target
+ *      </a>
+ */
+typedef struct __attribute__((packed)) {
+    uint8_t type;               /**< option type */
+    uint8_t length;             /**< option length without the first two bytes */
+    uint8_t flags;              /**< unused */
+    uint8_t prefix_length;      /**< number of valid leading bits in the IPv& prefix */
+    ng_ipv6_addr_t target;      /**< IPv& prefix, address or multicast group */
+} ng_rpl_opt_target_t;
+
+/**
+ * @brief DODAG Information Solicitation
+ * @see <a href="https://tools.ietf.org/html/rfc6550#section-6.7.8">
+ *          Transit Information
+ *      </a>
+ */
+typedef struct __attribute__((packed)) {
+    uint8_t type;               /**< option type */
+    uint8_t length;             /**< option length without the first two bytes */
+    uint8_t e_flags;            /**< external flag indicates external routes */
+    uint8_t path_control;       /**< limits the number of DAO parents */
+    uint8_t path_sequence;      /**< increased value for route updates */
+    uint8_t path_lifetime;      /**< lifetime of routes */
+} ng_rpl_opt_transit_t;
+
+typedef struct ng_rpl_dodag ng_rpl_dodag_t;
+typedef struct ng_rpl_parent ng_rpl_parent_t;
+
+/**
+ * @brief Parent representation
+ */
+struct ng_rpl_parent {
+    ng_rpl_parent_t *next;          /**< pointer to the next parent */
+    uint8_t state;                  /**< 0 for unsued, 1 for used */
+    ng_ipv6_addr_t addr;            /**< link-local IPv6 address of this parent */
+    uint16_t rank;                  /**< rank of the parent */
```

Comment:
In general, I am not in favor of using the byte-ordered types on structs, which will not be used for network communication.. This will make reading/writing from/to them more complex, because one would need to use the related `byteorder_*` functions throughout the code